tests/strands/agent/test_conversation_manager.py

```diff
@@ -246,3 +246,438 @@ def test_null_conversation_does_not_restore_with_incorrect_state():
+                        "toolUseId": "4",
+                        "name": "test_tool",
+                        "input": {"query": "fourth"},
+                    }
+                }
+            ],
+        },
+        # Response 5: Tool call
+        {
+            "role": "assistant",
+            "content": [
+                {
+                    "toolUse": {
+                        "toolUseId": "5",
+                        "name": "test_tool",
+                        "input": {"query": "fifth"},
+                    }
+                }
+            ],
+        },
+        # Final response: Text
+        {
+            "role": "assistant",
+            "content": [{"text": "All done!"}],
+        },
+    ]
+
+
+class TestPerTurnParameter:
+    """Tests for the per_turn parameter in SlidingWindowConversationManager."""
+
+    def test_per_turn_false_default(self):
+        """Test that per_turn defaults to False and no hooks are registered."""
+        manager = SlidingWindowConversationManager()
+        assert hasattr(manager, "per_turn")
+        assert manager.per_turn is False
+
+        # Verify register_hooks exists but doesn't register when per_turn=False
+        from strands.hooks.registry import HookRegistry
+
+        registry = HookRegistry()
+        if hasattr(manager, "register_hooks"):
+            manager.register_hooks(registry)
+            assert not registry.has_callbacks()
+
+    def test_per_turn_explicit_false(self):
+        """Test that per_turn=False explicitly works."""
+        manager = SlidingWindowConversationManager(per_turn=False)
+        assert manager.per_turn is False
+
+    def test_per_turn_true(self):
+        """Test that per_turn=True can be set."""
+        manager = SlidingWindowConversationManager(per_turn=True)
+        assert manager.per_turn is True
+
+    def test_per_turn_integer(self):
+        """Test that per_turn accepts integer values."""
+        manager = SlidingWindowConversationManager(per_turn=3)
+        assert manager.per_turn == 3
+
+    def test_per_turn_invalid_zero(self):
+        """Test that per_turn=0 raises ValueError."""
+        with pytest.raises(ValueError):
+            SlidingWindowConversationManager(per_turn=0)
+
+    def test_per_turn_invalid_negative(self):
+        """Test that negative per_turn values raise ValueError."""
+        with pytest.raises(ValueError):
+            SlidingWindowConversationManager(per_turn=-1)
+
+
+class TestHookRegistration:
+    """Tests for hook registration when per_turn is enabled."""
+
+    def test_register_hooks_exists(self):
+        """Test that register_hooks method exists."""
+        manager = SlidingWindowConversationManager()
+        assert hasattr(manager, "register_hooks")
+        assert callable(manager.register_hooks)
+
+    def test_register_hooks_with_per_turn_false(self):
+        """Test that no callbacks are registered when per_turn=False."""
+        from strands.hooks.registry import HookRegistry
```

Comment:
Import this at the top of the file, not per test

Remove these from all of the individual tests.  Same for `MockedModelProvider`.

Do a pass through and elevate any other local imports that can be elevated